### ISSUE TITLE
Production dependency not making it into ASAR node_modules

### ISSUE THREAD

**Issue by jlipps** [electron-userland/electron-builder#6145]
* **Electron-Builder Version**: 22.11.11

* **Node Version**: 14.17.4

* **Electron Version**: 13.1.8
* **Electron Type (current, beta, nightly)**: current

* **Target**: macOS

My application runs just fine using e.g. `npx electron .`. However when I bundle it using electron-builder, I got an error message preventing app load, saying that the 'request' library could not be found. The 'request' library is a dependency of one of my production dependencies ('request-promise').

I checked inside the ASAR file produced by electron-builder, and sure enough, there is no 'request' directory inside, even though there is a 'request-promise' directory inside. I assume this is because NPM recently changed to hoist all deduped deps to the top level.

In the docs, I read that electron-builder only includes 'production dependencies'. I'm not sure what algorithm it uses to determine this, but could it be that there is a bug in conjunction with newer versions of NPM, where deps of deps hoisted to the top node modules folder are being excluded by electron-builder because it doesn't see them in the package.json as production deps?

And if so, is there a possible workaround where I can tell electron-builder to include *all* files under node_modules, and then I just make sure to only install the prod deps before building?

**Comment by jlipps** [comment]
As a counterpoint to my hypothesis though, there do seem to be lots of other non top-level deps in the ASAR. So what would be different about request?

**Comment by mmaietta** [comment]
If there's an issue with filtering the node modules, then it'll likely be an issue within app-builder-bin, which is written in Go.
What happens if you add `includeSubNodeModules: true` to your electron-builder config?

**Comment by jlipps** [comment]
Good suggestion, I did not see that option before. It worked! It also increased the uncompressed .app size about 5x, to a horrendous ~1G. Seems to include all dev deps. Is there a way to get more fine-grained here with this option, or should I create an issue for app-builder-bin?

**Comment by mmaietta** [comment]
Woowza, that's too large. 
Yeah, it might be a bug in the upstream mom module. You mentioned that something changed recently in NPM deduping?
Can you post your package.json here or create a sample repo with electron-quick-start?

**Comment by jlipps** [comment]
Yes, it is all in the main branch of [this repo](https://github.com/appium/appium-inspector/). Assuming you're on a mac, the steps to repro should be as easy as:

1. Clone repo and git submodule update
2. `npm install`
3. `npm run build`
4. `npx electron-builder build --publish never`

Then the .app will be in the `release/mac` dir.

**Comment by jlipps** [comment]
Did you have a chance to dig into this at all, @mmaietta?
